torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -1092,10 +1092,7 @@ at::Tensor XLANativeFunctions::cumsum(const at::Tensor& self, int64_t dim,
 // TODO(alanwaketan): Let's rewrite a without reusing other native functions.
 at::Tensor XLANativeFunctions::detach_copy(const at::Tensor& self) {
   TORCH_LAZY_FN_COUNTER("xla::");
-  auto new_tensor =
-      empty_symint(self.sym_sizes(), at::typeMetaToScalarType(self.dtype()),
-                   c10::nullopt, self.device(), c10::nullopt, c10::nullopt);
-  return set_(new_tensor, self);
+  return bridge::AtenFromXlaTensor(bridge::GetXlaTensor(self));
```

Comment:
Same as view_copy.

Reply:
Replace this with view_copy actually increase the memory footprint. So, the current implementation is actually better.

Reply:
hmm, is this real a copy? Aren't we just return the same XLATensor back to the caller?

Reply:
for example if someone call `mark_sharding` on this XLATensor, I think it will affect both new and old tensors. I don't think that's the right behavior.

Reply:
Well, detach is never a copy. So I think it's okay: https://pytorch.org/docs/stable/generated/torch.Tensor.detach.html

Just like view_copy, we also don't copy anything. Instead, we make a view node. Therefore, I think here we just need to respect the original behavior  and no needs to deal with "copying".